### PR TITLE
Operative Endpoint refactoring

### PR DESCRIPTION
**Pull request by anmolanmol1234**
Remove unwanted params and refactor the code for OperativeEndpoint.

### CODE REVIEW

File: hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java [hunk 1]
```diff
@@ -1601,7 +1601,7 @@ public void rename(final Path source, final Path destination,
   }
 
   private Boolean isCreateOperationOnBlobEndpoint() {
-    return !OperativeEndpoint.isIngressEnabledOnDFS(prefixMode, abfsConfiguration);
+    return !OperativeEndpoint.isIngressEnabledOnDFS(getPrefixMode(), abfsConfiguration);
```
Comment (anujmodi2021) [inline hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java:1604]: I was not able to follow why we need to send prefix mode as a separate argument here??
As per offline discussion you mentioned that 

"The intent on sending the prefixMode as separate param was for the fallback mechanism we implemeted in append and flush flow. If a request we fallback to DFS we don't want to change the global prefix mode only at abfsOutputStream level. Hence I had added prefixMode as a separate param"

Is there a scenario where we are changing the prefix mode locally for a call??

Comment (anmolanmol1234) [reply hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java:1604]: yes when append over blob fails with a 409 blob operation not supported, that means create was done over dfs endpoint and hence append and flush also need to go over dfs. So in AbfsOuputStream class we change the local variable to have prefix mode as dfs.

Comment (anujmodi2021) [reply hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java:1604]: Makes sense...
Thanks

**Review by anujmodi2021 — APPROVED** [review]

**Review by sreeb-msft — APPROVED** [review]
LGTM!

**Pull request merged**